scripts/karma.coffee

```diff
@@ -138,19 +138,29 @@ class Karma
     sorted = @sort()
     sorted.slice(-n).reverse()
 
+  notify: (msg, subject, user, verb) ->
+    msgPlace = switch msg.message.rawMessage.channel._modelName
+      when "DM" then "in a direct message"
+      when "Channel" then "in the channel \"#{msg.message.rawMessage.channel.name}\""
+      when "Group" then "in the private channel \"#{msg.message.rawMessage.channel.name}\""
+      else "by some means of sorcery"
+    return "#{user.name} #{verb} #{subject} (Karma: #{this.get(subject)}) #{msgPlace}"
+
 module.exports = (robot) ->
   karma = new Karma robot
   robot.karma = karma
   cronJob = require('cron').CronJob
   new cronJob('0 01 01 * * *', karma.clearAllowances, null, true, 'America/Chicago', karma)
   allow_self = process.env.KARMA_ALLOW_SELF or "true"
+  notify_channel = process.env.KARMA_NOTIFY_CHANNEL or "karma"
```

Comment:
I'm still not crazy about this, I think if we're going to have a default, it should default to whatever the test-karma channel is, not the "prod". But I think it would be ideal if the whole notify thing just didn't work at all if the env variable isn't set.

Reply:
After thinking about this, I agree. Can we move notifications to skip if there isn't a variable defined?